tests/monotouch-test/System.Net.Http/MessageHandlers.cs

```diff
@@ -120,6 +120,84 @@ public void TestNSUrlSessionHandlerCookies ()
 			Assert.That (managedCookies.First (), Is.StringStarting ("cookie=chocolate-chip;"), $"Managed Cookie Value");
 		}
+
+		// ensure that we can use a cookie container to set the cookies for a url
+		[Test]
+		public void TestNSUrlSessionHandlerCookieContainer ()
+		{
+			var url = NetworkResources.Httpbin.CookiesUrl;
+			var cookie = new Cookie ("cookie", "chocolte-chip");
+			var cookieContainer = new CookieContainer ();
+			cookieContainer.Add (new Uri (url), cookie);
+
+			string managedCookieResult = null;
+			string nativeCookieResult = null;
+			Exception ex = null;
+			var completed = false;
+
+			TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () => {
+				try {
+					var managedHandler = new HttpClientHandler () {
+						AllowAutoRedirect = false,
+						CookieContainer = cookieContainer,
+					};
+					var managedClient = new HttpClient (managedHandler);
+					var managedResponse = await managedClient.GetAsync (url);
+					managedCookieResult = await managedResponse.Content.ReadAsStringAsync ();
+					
+					var nativeHandler = new NSUrlSessionHandler () {
+						AllowAutoRedirect = true,
+						CookieContainer = cookieContainer,
+					};
+					var nativeClient = new HttpClient (nativeHandler);
+					var nativeResponse = await nativeClient.GetAsync (url);
+					nativeCookieResult = await nativeResponse.Content.ReadAsStringAsync ();
+				} catch (Exception e) {
+					ex = e;
+				} finally {
+					completed = true;
+				}
+			}, () => completed);
+
+			Assert.IsNull (ex, "Exception");
+			Assert.IsNotNull (managedCookieResult, "Managed cookies result");
+			Assert.IsNotNull (nativeCookieResult, "Native cookies result");
+			Assert.AreEqual (managedCookieResult, nativeCookieResult, "Cookies");
+		}
+
+		// ensure that the Set-Cookie headers do update the CookieContainer
+		[Test]
+		public void TestNSurlSessionHandlerCookieContainerSetCookie ()
+		{
+			var url = NetworkResources.Httpbin.GetSetCookieUrl ("cookie", "chocolate-chip");
+			var cookieContainer = new CookieContainer ();
+
+			string nativeCookieResult = null;
+			Exception ex = null;
+			var completed = false;
+
+			TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () => {
+				try {
+		
+					var nativeHandler = new NSUrlSessionHandler () {
+						AllowAutoRedirect = true,
+						CookieContainer = cookieContainer,
+					};
+					var nativeClient = new HttpClient (nativeHandler);
+					var nativeResponse = await nativeClient.GetAsync (url);
+					nativeCookieResult = await nativeResponse.Content.ReadAsStringAsync ();
+				} catch (Exception e) {
+					ex = e;
+				} finally {
+					completed = true;
+				}
+			}, () => completed);
+
+			Assert.IsNull (ex, "Exception");
+			Assert.IsNotNull (nativeCookieResult, "Native cookies result");
+			var cookiesFromServer = cookieContainer.GetCookies (new Uri (url));
+			Assert.AreEqual (1, cookiesFromServer.Count, "Cookies received from server.");
+		}
```

Comment:
It would worth checking the name and the value of the actual cookie as well.

Reply:
Valid point. Merge and will update the test :)